activerecord/lib/active_record/counter_cache.rb

```diff
@@ -113,6 +113,7 @@ def reset_counters(id, *counters, touch: nil)
       #   #    `updated_at` = '2016-10-13T09:59:23-05:00'
       #   #  WHERE id IN (10, 15)
       def update_counters(id, counters)
+        id = [id] if composite_primary_key? && id.is_a?(Array) && !id[0].is_a?(Array)
```

Comment:
@casperisfine So after a small tweak this is the only place where we end up with a "not so pretty" handling. How would you feel about merging this as a bugfix but we will still work on a solution to make sure we get rid of `composite_primary_key? ` branches and everything that comes with it in a separate PR. Because even if we come up with something I'd prefer to ship it separately as this bugfix will need to be backported while a refactoring can be released in the next major